### ISSUE TITLE
learn anything efficiently - a structured approach

### ISSUE THREAD

**Issue by srele96** [srele96/sk-experiments#181]
## What

How to efficiently learn a tool:

1. read the tool's documentation
2. find out what problem it solves
3. think of a problem that can be solved using that tool
4. make the problem my own and find what problem I want to solve
5. create a problem definition
6. look for the ways a tool solves that problem
7. find a solution and implement it using the tool

Alternatively

1. read the tool's documentation
2. find out how the tool work
  (for example, ci/cd runs stuff on their machine, how can I use that?)
3. think of a problem that can be solved using that tool
4. make the problem my own and find what problem I want to solve
5. create a problem definition
6. look for the ways a tool solves that problem
7. find a solution and implement it using the tool

Basically, try:

1. figure out what is the concept of the library
2. think about the concept, how can I use it
3. attack the library from the creator side, for example (I know the concept, think of problems think what can I make with it, why do I want to make it, etc... then try to come up with a problem and use library to solve it, basically find a concept and then try to use library to solve the concept)

## Why

Learn anything in a really fast and efficient manner.

## How

Try it for some domains I am not familiar with.
